cmd/util/ledger/util/registers_util.go

```diff
@@ -0,0 +1,106 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/util/registers"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func NewByAccountRegistersFromPayloadAccountGrouping(
+	payloadAccountGrouping *PayloadAccountGrouping,
+	nWorker int,
+) (
+	*registers.ByAccount,
+	error,
+) {
+	g, ctx := errgroup.WithContext(context.Background())
+
```

Comment:
do we need to adjust nWorker with the account count?

nWorker = min(nWorker, payloadAccountGrouping.Len())

Reply:
> do we need to adjust nWorker with the account count?

Probably not needed because we have millions of accounts and nWorkers will be closer to only about 100.

However, it doesn't hurt to add checks here :+1: so I added sanity checks for number of accounts and nWorkers.